src/Request/Init.php

```diff
@@ -142,6 +157,27 @@ public function __construct(string $service, $securityPacket, string $secret, $r
         $this->securityPacket['signature'] = $this->generateSignature();
     }
 
+    /**
+     * Get or create a signature generator for authentication
+     *
+     * @return SignatureInterface
+     * @throws \Exception
+     */
+    protected function getSignatureGenerator(): SignatureInterface
+    {
+        if (!isset($this->signatureGenerator)) {
+            $this->setSignatureGenerator(SignatureFactory::getSignatureGenerator(
+                self::DEFAULT_SIGNATURE_VERSION
```

Comment:
I think we should make this parameter optional, so the factory is in sole control of what the default is.

Reply:
Good shout.